# Task 05 - Descriptive Statistics & LLM Validation
**Dataset:** Syracuse Men's Soccer 2024

This notebook shows:
1. Loading the dataset
2. Python analysis for each question
3. LLM answers for the same questions
4. Comparison and validation

In [2]:
import pandas as pd

# Load dataset
file_path = 'syracuse_mens_soccer_2024_full.xlsx'
df_players = pd.read_excel(file_path, sheet_name='Players')
df_goalies = pd.read_excel(file_path, sheet_name='Goalies')
df_team_totals = pd.read_excel(file_path, sheet_name='Team Totals')

print('Players DataFrame:')
display(df_players.head())
print('\nTeam Totals DataFrame:')
display(df_team_totals)

Players DataFrame:


Player  GP  G  A  PTS  SH    SH%  SOG   SOG% YC-RC  GW PK-ATT
0   Threadgold, Gabe  17  4  3   11  18  0.222   12  0.667   1-0   1    0-0
1    Acquah, Michael  16  1  6    8  24  0.042    9  0.375   2-1   1    0-0
2     Omeze, Chimere  16  3  1    7  17  0.176    8  0.471   7-0   2    0-0
3      Burko, Daniel  16  3  0    6  13  0.231    6  0.462   0-0   0    0-0
4  Fortier, Kristjan  14  2  0    4   6  0.333    4  0.667   3-0   1    0-0


Team Totals DataFrame:


Player  GP   G   A  PTS   SH    SH%  SOG   SOG% YC-RC  GW PK-ATT
0      Team Total  17  25  20   70  233  0.107  100  0.429  40-2   7    2-2
1  Opponent Total  17  25  21   71  204  0.123   82  0.402  37-0   7    3-3

## 1️⃣ Average Goals per Game

In [3]:
total_goals = df_players['G'].sum()
games_played = df_players['GP'].max()
avg_goals_per_game = total_goals / games_played
print('Average Goals per Game:', round(avg_goals_per_game, 2))

Average Goals per Game: 1.47


**LLM Answer:**
> The Syracuse men’s soccer team averaged about 1.47 goals per game in the 2024 season, reflecting a total of 25 goals over 17 matches.

**Comparison:** ✅ Matches Python analysis.

## 2️⃣ Top 3 Players by Total Points (G+A)

In [4]:
df_players['Points'] = df_players['G'] + df_players['A']
top3_points = df_players[['Player','Points','G','A']].sort_values('Points', ascending=False).head(3)
display(top3_points)

Player  Points  G  A
0  Threadgold, Gabe       7  4  3
1   Acquah, Michael       7  1  6
2    Omeze, Chimere       4  3  1

**LLM Answer:**
1. Gabe Threadgold – 7 points (4G + 3A)
2. Michael Acquah – 7 points (1G + 6A)
3. Chimere Omeze – 4 points (3G + 1A)

**Comparison:** ✅ Matches Python output.

## 3️⃣ Player with Highest Shot Accuracy (≥5 shots)

In [5]:
df_players['ShotAccuracy'] = df_players['SOG'] / df_players['SH']
top_accuracy = df_players[df_players['SH']>=5].sort_values('ShotAccuracy', ascending=False).head(1)
display(top_accuracy[['Player','SH','SOG','ShotAccuracy']])

Player  SH  SOG  ShotAccuracy
0  Threadgold, Gabe  18   12      0.666667

**LLM Answer:**
> Gabe Threadgold – 18 shots, 12 on target, 66.7% accuracy

**Comparison:** ✅ Matches Python output.

## 4️⃣ Highest Impact per 90 mins (G+A per 90)

In [6]:
df_players['MIN'] = df_players['GP'] * 90
df_players['G+A_per_90'] = (df_players['G'] + df_players['A']) / (df_players['MIN']/90)
top_impact = df_players.sort_values('G+A_per_90', ascending=False).head(3)
display(top_impact[['Player','GP','G','A','G+A_per_90']])

Player  GP  G  A  G+A_per_90
12    Gradus, Michal   2  1  0    0.500000
1    Acquah, Michael  16  1  6    0.437500
0   Threadgold, Gabe  17  4  3    0.411765

**LLM Answer:**
1. Michal Gradus – 0.50 G+A/90
2. Michael Acquah – 0.44 G+A/90
3. Gabe Threadgold – 0.41 G+A/90

**Comparison:** ✅ Matches Python output.

## 5️⃣ Game-Winning Contribution

In [7]:
df_players['GW_Contribution'] = df_players['GW']
top_gw = df_players.sort_values('GW_Contribution', ascending=False).head(3)
display(top_gw[['Player','G','GW_Contribution']])

Player  G  GW_Contribution
2     Omeze, Chimere  3                2
0   Threadgold, Gabe  4                1
4  Fortier, Kristjan  2                1

**LLM Answer:**
1. Chimere Omeze – 2 GW goals
2. Gabe Threadgold – 1 GW goal
3. Kristjan Fortier – 1 GW goal

**Comparison:** ✅ Matches Python output.

## 6️⃣ Discipline vs Performance (G+A per Card)

In [8]:
df_players[['YC','RC']] = df_players['YC-RC'].str.split('-', expand=True).astype(int)
df_players['TotalCards'] = df_players['YC'] + df_players['RC']
df_players['Contribution_per_Card'] = (df_players['G'] + df_players['A']) / df_players['TotalCards'].replace(0,1)
top_disciplined = df_players.sort_values('Contribution_per_Card', ascending=False).head(3)
display(top_disciplined[['Player','G','A','TotalCards','Contribution_per_Card']])

Player  G  A  TotalCards  Contribution_per_Card
0  Threadgold, Gabe  4  3           1                    7.0
3     Burko, Daniel  3  0           0                    3.0
7    Smith, Braedon  1  2           1                    3.0

**LLM Answer:**
1. Gabe Threadgold – 7 G+A per card
2. Daniel Burko – 3 G+A per card
3. Braedon Smith – 3 G+A per card

**Comparison:** ✅ Matches Python output.

## 7️⃣ Coaching Question: Attack or Defense Focus

In [9]:
team_totals = df_team_totals[df_team_totals['Player'] == 'Team Total'].iloc[0]
opp_totals = df_team_totals[df_team_totals['Player'].str.contains('Opponent', case=False)].iloc[0]

team_goals = team_totals['G']
team_shots = team_totals['SH']
team_sog = team_totals['SOG']

opp_goals = opp_totals['G']
opp_shots = opp_totals['SH']
opp_sog = opp_totals['SOG']

team_sh_pct = team_goals / team_shots
opp_sh_pct = opp_goals / opp_shots
team_sog_pct = team_sog / team_shots
opp_sog_pct = opp_sog / opp_shots

print(f'Team Shot%: {team_sh_pct:.3f}, Opponent Shot%: {opp_sh_pct:.3f}')
print(f'Team SOG%:  {team_sog_pct:.3f}, Opponent SOG%: {opp_sog_pct:.3f}')

focus = 'Defense' if opp_goals > team_goals or opp_sh_pct > team_sh_pct else 'Attack'
print(f'\nRecommended Focus Area for Next Season: {focus}')

Team Shot%: 0.107, Opponent Shot%: 0.123
Team SOG%:  0.429, Opponent SOG%: 0.402

Recommended Focus Area for Next Season: Defense


**LLM Answer:**
> Focus on Defense. Syracuse conceded as many goals as they scored, and opponent shot efficiency was slightly higher.

**Comparison:** ✅ Matches Python-based reasoning.

## 8️⃣ Most Efficient Scorer (Goals per Shot)

Let's determine which player had the highest goal conversion rate (i.e., goals per shot) among those who took at least 5 shots.

In [10]:
# Filter players with at least 5 shots
df_players['Goals_per_Shot'] = df_players['G'] / df_players['SH']
efficient_scorers = df_players[df_players['SH'] >= 5].sort_values('Goals_per_Shot', ascending=False).head(3)
display(efficient_scorers[['Player', 'G', 'SH', 'Goals_per_Shot']])


Player  G  SH  Goals_per_Shot
4  Fortier, Kristjan  2   6        0.333333
3      Burko, Daniel  3  13        0.230769
0   Threadgold, Gabe  4  18        0.222222

🧠 LLM Answer

Kristjan Fortier had the highest goal conversion rate among frequent shooters, scoring 2 goals from just 6 shots (33.3%). This suggests he's a highly efficient finisher and may be underutilized in attack.

✅ Comparison

✔️ LLM matched Python output correctly
✔️ Provided accurate reasoning and interpretation
✔️ Value added: recognizes possible underuse of Fortier despite efficiency

## 9️⃣ Goalkeeper Efficiency (Saves per Goal Conceded)

This analysis helps us evaluate the performance of the team’s goalkeeper(s). A higher saves-to-goals-conceded ratio indicates stronger shot-stopping ability.

In [13]:
df_goalies['Saves_per_GA'] = df_goalies['Saves'] / df_goalies['GA']
goalie_eff = df_goalies.sort_values('Saves_per_GA', ascending=False)
display(goalie_eff[['Player', 'MIN', 'GA', 'Saves', 'Saves_per_GA']])


Player     MIN  GA  Saves  Saves_per_GA
1  Smith, Jason    70.6   1      3      3.000000
0    Hut, Tomas  1459.4  24     52      2.166667

🧠 LLM Answer:

"Jason Smith has the highest shot-stopping efficiency with 3.0 saves per goal conceded, though based on a limited sample size of 70.6 minutes.
Tomas Hut played the majority of the season and maintained a strong 2.17 saves per goal, a solid performance given the workload."

## 🔟 Underrated Efficiency: Points per Shot (PTS/SH)

Objective:
To uncover players who make the most of their opportunities, we’ll calculate how many points (G+A) each player generates per shot. This reveals under-the-radar efficiency in scoring contribution.

In [14]:
# Calculate Points per Shot
df_players['PTS_per_SH'] = df_players['PTS'] / df_players['SH']
efficient_players = df_players[df_players['SH'] >= 5].sort_values('PTS_per_SH', ascending=False).head(5)
display(efficient_players[['Player', 'PTS', 'SH', 'PTS_per_SH']])


Player  PTS  SH  PTS_per_SH
4       Fortier, Kristjan    4   6    0.666667
0        Threadgold, Gabe   11  18    0.611111
3           Burko, Daniel    6  13    0.461538
10  Cutler-DeJesus, Andre    3   7    0.428571
2          Omeze, Chimere    7  17    0.411765

Kristjan Fortier stands out as the most efficient contributor, with 0.667 points per shot — meaning nearly 2 out of every 3 shots led to a goal or assist.
Gabe Threadgold and Daniel Burko follow, both showing solid efficiency with fewer wasted opportunities.
This metric highlights players who may be underutilized given their conversion impact.